### PR TITLE
13: Created CreateEvent Component

### LINKED ISSUE #13: Create CreateEvent component
Make a component called CreateEvent. This will be used by admins to create a volunteer event to (eventually) be added to the calendar feature. **This task does not include implementing the functionality to display the modal (the "pop-up" action). Focus on the component's structure and styles**

Figma: https://www.figma.com/file/Z4ehMEyfixqSX5K5EJZ9ID/ccuss-project?type=design&node-id=128-400&mode=design&t=VSaK84qfttjxs3Qc-0 

Image

- [x] Event name field
- [x] Event date (use some datepicker library, maybe MaterialUI)
- [x] Event description field
- [x] questions useState, and leave a space for the AddQuestions component at the bottom
- [x] POST request to create event with data in the fields.

Notes/tips:
- You will need to create a `[questions, setQuestions] = useState([]);` statement. This `questions` array and `setQuestions` function will be passed to the nested `AddQuestions` component later on. It will also be used to do a POST to VolunteerForms later on.
- Ignore the Event.form attribute in your Event POST data for now -- it's an optional field in the Event schema.
- We are removing the start and end time fields seen in the Figma design
- Make sure your dates get formatted correctly before you POST
- Use the dimensions in the Figma as this will be a modal popup

### PR DESCRIPTION
**Pull request by kaseyliu**
## Developer: {Kasey Liu, Amanda Chan}

Closes #13 

### Pull Request Summary

We created a CreateEvent form including fields like title, start time, end time, date, and description. There is also a button for the AddQuestion component, but for now there is no functionality to it. We then created a POST request to create an event with all the given data. We made the form and roles default for now, as they were not fields specified on the design, but they were specified in the schema. 

### Modifications

CreateEvent.tsx: Created fields for title, start time, end time, date picket for date, and description. made a POST request to create an event with this data.

### Testing Considerations

Tested that the event shows up in MongoDB and that the fields are filled out correctly with console.log for the data filled out.

### Pull Request Checklist

- [x] Code is neat, readable, and works
- [x] Comments are appropriate
- [x] The commit messages follows our [guidelines](https://h4i.notion.site/Conventional-Commits-593452ad1179489399ad3bd696ef772a)
- [x] The developer name is specified
- [x] The summary is completed
- [x] Assign reviewers

### Screenshots/Screencast

website:
Screenshot 2024-02-29 at 3 31 12 PM

console:
Screenshot 2024-02-29 at 2 31 04 PM

mongo:
Screenshot 2024-02-29 at 2 31 21 PM

### CODE REVIEW

File: package.json
Comment (andrewtchan) [inline package.json:1]: Please revert these changes to package.json and package-lock.json so it matches what's on dev. You can re-add dayjs.

Comment (kaseyliu) [reply package.json:1]: Just reverted it I think, let me know if it is ok now!

Comment (andrewtchan) [reply package.json:1]: Yes I see it now, thanks. Sometimes hard to keep track of these package things bc your local node_modules may not match what's on the package.json unless you do a clean install.

File: src/app/components/CreateEvent/CreateEvent.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        setDescription('');
+        setQuestions([]);
+        setRoles([]);
+        setLocation('default location')
+    }
+
+    const handleSubmit = async () => {
+        // POST each role
+
+        // compile questions into VolunteerForm, POST form
```
Comment (andrewtchan) [inline src/app/components/CreateEvent/CreateEvent.tsx:42]: We will need to add this functionality (different PR).
